securedrop/manage.py

```diff
@@ -313,7 +313,7 @@ def translate_messages(args):
                    sources=" ".join(args.source)))
 
         changed = subprocess.call("git diff --quiet {}".format(messages_file),
```

Comment:
Since we're calling this with `--quiet`, can we not disable shell here too?

Reply:
Good call, unfortunately the change does not appear trivial (to me). Changing that line to:
```
changed = subprocess.call(['git', 'diff', '--quiet', messages_file])
```
makes `tests/test_i18n.py::TestI18N::test_i18n` fail with `OSError: [Errno 2] No such file or directory`. Any ideas on what's going on?

Reply:
not really
<pre>
>>> import subprocess
>>> subprocess.call(['git', 'rev-parse', 'HEAD'])
f0c32641464892897f172bd601e3522cbba518f3
</pre>
so `message_file` must be not found but ... how ? I suspect this is unrelated. Does it work if you revert back to using shell=True ?

Reply:
It could be an absolute path thing. Maybe `path.abspath(messages_file)`?

Reply:
@dachary it does indeed pass with:
```
changed = subprocess.call(['git', 'diff', '--quiet', messages_file], shell=True)
```

Reply:
this is really wierd, what kind of side effect can shell=True have to make a difference ?

Reply:
Using `os.path.abspath(messages_file)` still throws the error. Given the `sh` function in the file, I suggest we keep the nosec annotation.

Reply:
Totally fair. Can you open a ticket to investigate this? Could be something a newbie could look in to too since it's is only a couple line change.

Reply:
https://github.com/freedomofpress/securedrop/issues/3071